### PR TITLE
*: skip a case in TestClusterTables/ForClusterServerInfo on mac M1

### LINKED ISSUE #29757: `TestClusterTables/ForClusterServerInfo` fails on apple M1
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

`make test`

### 2. What did you expect to see? (Required)

all cases pass
### 3. What did you see instead (Required)

```
--- FAIL: TestClusterTables (2.11s)
    --- FAIL: TestClusterTables/ForClusterServerInfo (1.53s)
        cluster_tables_serial_test.go:154:
                Error Trace:    cluster_tables_serial_test.go:154
                Error:          Not equal:
                                expected: set.StringSet{"cpu":struct {}{}, "disk":struct {}{}, "memory":struct {}{}, "net":struct {}{}}
                                actual  : set.StringSet{"disk":struct {}{}, "memory":struct {}{}, "net":struct {}{}}

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,4 +1,2 @@
                                -(set.StringSet) (len=4) {
                                - (string) (len=3) "cpu": (struct {}) {
                                - },
                                +(set.StringSet) (len=3) {
                                  (string) (len=4) "disk": (struct {}) {
                Test:           TestClusterTables/ForClusterServerInfo
                Messages:       sql: select * from information_schema.CLUSTER_HARDWARE;
```

### 4. What is your TiDB version? (Required)

master branch
golang 1.17.2
apple M1

I think the root cause is https://github.com/pingcap/sysutil/blob/e5c5906ed1e736d38565158bc6da9e3f2ee07034/hardware_info.go#L18. This call will fail on apple M1 due to https://github.com/shirou/gopsutil/issues/1000.

### PR DESCRIPTION
**Pull request by zhangyangyu**
### What problem does this PR solve?

Issue Number: close #29757 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: infoschema/cluster_tables_serial_test.go [hunk 1]
```diff
@@ -125,13 +127,15 @@ func SubTestForClusterServerInfo(s *clusterTablesSuite) func(*testing.T) {
...
-				skipOnOS: "windows",
+				skipOnDist: "(windows/.+)",
 			},
 		}
 
 		for _, cas := range cases {
-			if cas.skipOnOS == runtime.GOOS {
-				continue
+			if cas.skipOnDist != "" {
+				if matched, _ := regexp.MatchString(cas.skipOnDist, runtime.GOOS+"/"+runtime.GOARCH); matched {
```
Comment (tisonkun) [inline infoschema/cluster_tables_serial_test.go:136]: replace with `&&`?

Comment (zhangyangyu) [reply infoschema/cluster_tables_serial_test.go:136]: Ahh, seems not possible to `&&` with a multivalue return?

Comment (tisonkun) [reply infoschema/cluster_tables_serial_test.go:136]: Make sense. I suggest you try to add a utility for detecting specific platforms. Especially for Windows and Apple M1. And thus avoid these inline handmade stuff.

Comment (tisonkun) [reply infoschema/cluster_tables_serial_test.go:136]: It looks quite brittle you bring a new pattern to concatenate from OS + ARCH here and place some regexes above.

Comment (zhangyangyu) [reply infoschema/cluster_tables_serial_test.go:136]: I agree. I've spent some time trying to figure out a better solution but failed. Maybe use a function to hide the regexes.

**Review by tisonkun — APPROVED** [review]
/cc @xhebox

**Review by xhebox — APPROVED** [review]

**Pull request merged**